Recipes/Recipes/All Recipes/Presenter/AllRecipesPresenter.swift

```diff
@@ -0,0 +1,176 @@
+            } else if sortedTime == .timeLow {
+                view?.buttonTimeState(color: "background01", image: "filterHigh")
+                sortedTime = .timeHigh
+                sortRecipes(category: category)
+            } else if sortedTime == .timeHigh {
+                sortedTime = .none
+                view?.buttonTimeState(color: "background06", image: "filterIcon")
+                sortRecipes(category: category)
+            }
+        }
+}
+
+extension AllRecipesPresenter: RecipeProtocol {
+
+    func sortRecipes(category: [Recipes]) {
+        let defaultRecipes = Recipes.allRecipes
+        var sorted = category
+
+        let sortCalories: ((Recipes, Recipes) -> Bool)?
+        switch sortedCalories {
+        case .caloriesLow:
+            sortCalories = { Int($0.caloriesTitle) ?? 0 < Int($1.caloriesTitle) ?? 0 }
+        case .caloriesHigh:
+            sortCalories = { Int($0.caloriesTitle) ?? 0 > Int($1.caloriesTitle) ?? 0 }
+        default:
+            sortCalories = nil
+        }
+
+        let sortTime: ((Recipes, Recipes) -> Bool)?
+        switch sortedTime {
+        case .timeLow:
+            sortTime = { Int($0.cookingTimeTitle) ?? 0 < Int($1.cookingTimeTitle) ?? 0 }
+        case .timeHigh:
+            sortTime = { Int($0.cookingTimeTitle) ?? 0 > Int($1.cookingTimeTitle) ?? 0 }
+        default:
+            sortTime = nil
+        }
+
+        sorted = category.sorted { lhs, rhs in
+            if let sortCalories = sortCalories, let sortTime = sortTime {
+                    if lhs.caloriesTitle == rhs.caloriesTitle {
+                        return sortTime(lhs, rhs)
+                    } else {
+                        return sortCalories(lhs, rhs)
+                    }
+                } else if let sortCalories = sortCalories {
+                    return sortCalories(lhs, rhs)
+                } else if let sortTime = sortTime {
+                    return sortTime(lhs, rhs)
+                }
+                return false
+            }
+
+        view?.sortViewRecipes(recipes: sorted)
+        self.sorted = sorted
+    }
+    
+    func checkSearch() -> [Recipes] {
+        if isSearching {
+            return searchNames
+        } else {
+            return sorted
+        }
+    }
+    
+    func startSearch() {
+        isSearching = true
+    }
+    
+    func stopSearch() {
+        isSearching = false
+    }
+    
+    func searchRecipes(text: String) {
+        guard !text.isEmpty else {
+            isSearching = false
+            searchNames = []
+            view?.reloadTableView()
+            return
+        }
+        isSearching = true
+        searchNames = recipes.filter { $0.titleRecipies.lowercased().contains(text.lowercased()) }
+        view?.reloadTableView()
+    }
+    
+    func goToCategory() {
+        view?.goToTheCategory()
+    }
+    
+    func getCategoryTitle() {
+//        view?.setTitle(<#T##title: String##String#>)
```

Comment:
```suggestion
//        view?.setTitle(<#T##title: String##String#>)
```
коменты не допустимы